frontend/src/HomePage.jsx

```diff
@@ -12,34 +15,93 @@ import NewBoardForm from "./components/NewBoardForm/NewBoardForm";
 
 export default function HomePage() {
   const [newBoardFormOpened, setNewBoardFormOpened] = useState(false);
-  const [boardList, setBoardList] = useState(boards);
+  const [boardList, setBoardList] = useState([]);
+  const [boardListCopy, setBoardListCopy] = useState([]); //need this extra copy so that when we filter we dont alter and lose original board list
+  const filterType = useRef('all');
 
-  function handleSearch(searchInput) {}
+  function handleSearch(searchInput) {
+    const searchTerm = searchInput.toLowerCase();
 
-  function handleClear() {}
+    const searchResults = boardListCopy.filter(
+      (board) =>
+        (board.category === filterType.current || filterType.current === 'all') &&
+        board.title.toLowerCase().includes(searchTerm)
+    );
+    setBoardList(searchResults);
+  }
+
+  function handleClear() {
+    const cardsForPage = boardListCopy.filter((board) => board.category === filterType.current || filterType.current === 'all');
+    setBoardList(cardsForPage);
+  }
+
+  //function gets called in NewBoardForm component whenever user clicks submit. It recieves all the new board detaisl as parameters and passes them to createBoard which is the api fetch request
+  async function handleNewBoardSubmit(
+    event,
+    boardTitle,
+    boardCat,
+    boardAuthor
+  ) {
+    event.preventDefault(); //prevent default form behavior which makes it go away as soon as its submitted
+    const newBoard = await createBoard(boardTitle, boardAuthor, boardCat);
+    setBoardList((prevBoardList) => [...prevBoardList, newBoard]);
+    setBoardListCopy((prevBoardListCopy) => [...prevBoardListCopy, newBoard]);
+  }
+
+  //function gets passed all the way down to BoardCard so that it can be invoked when delete button is clicked.
+  async function handleDelete(deleteId) {
+    const deletedBoard = await deleteBoard(deleteId);
```

Comment:
Is a common practice return only an okay when you delete, then here you already have the id so you can filter it.